main.go

```diff
@@ -22,26 +28,74 @@ type rule struct {
 }
 
 type input struct {
-	Type     string `yaml:"type"`
-	ApiGroup string `yaml:"apiGroup"`
-	Version  string `yaml:"version"`
-	Resource string `yaml:"resource"`
+	// Name of the input, it could be referenced in the expression
+	Name string `yaml:"name"`
+	// Type of the input, it could be a kube GroupVersionResource or a file or a kube api path
+	Type        string `yaml:"type"`
+	ApiGroup    string `yaml:"apiGroup"`
+	Version     string `yaml:"version"`
+	Resource    string `yaml:"resource"`
+	SubResource string `yaml:"subResource,omitempty"`
+	Namespace   string `yaml:"namespace,omitempty"`
+	Path        string `yaml:"path"`
 }
 
 func main() {
 	validate()
 
 	r := read(ruleFile)
 
-	// collect dependencies (e.g., get file metadata information or
-	// Kubernetes resource information)
-	collect(r)
+	// collect dependencies (e.g., get file metadata information or Kubernetes resource information)
+	resultMap := collect(r)
 
-	// build a CEL scanning instance with the rule expression
+	declsList := []*expr.Decl{}
 
-	// run the CEL program with the inputs
+	for k, v := range resultMap {
+		fmt.Printf("Key: %s\nValue: %v\n", k, v)
+		// build a CEL environment with the rule expression
+		declsList = append(declsList, decls.NewVar(k, decls.Dyn))
+	}
+
+	// build a CEL environment with the rule expression
+	env, err := cel.NewEnv(
+		cel.Declarations(declsList...),
+	)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to create CEL environment: %s", err))
+	}
+
+	// compile the CEL expression
+	ast, issues := env.Compile(r.Expression)
+	if issues.Err() != nil {
+		panic(fmt.Sprintf("Failed to compile CEL expression: %s", issues.Err()))
+	}
+
+	evalVars := map[string]interface{}{}
+	for k, v := range resultMap {
+		evalVars[k] = toCelValue(v)
+	}
+
+	// evaluate the CEL program with the inputs
+	prg, err := env.Program(ast)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to create CEL program: %s", err))
+	}
+
+	out, _, err := prg.Eval(evalVars)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to evaluate CEL expression: %s", err))
```

Comment:
`panic: Failed to evaluate CEL expression: no such key: identityProviders`

If a key is not found we should print that it was not found and just fail the rule.

Reply:
Handeld in [71fd886](https://github.com/rhmdnd/cel-scanner-poc/pull/6/commits/71fd886d15ab022a5e9ca01e0f5ded347d4a0280)